File: check_riak_up.py
#!/usr/bin/python

import sys, os
def _nagios(hdr, msg, code):
    print '%s: %s' % (hdr, msg)
    return code

def critical(msg): return _nagios('CRITICAL', msg, 2)
def warning(msg): return _nagios('WARNING', msg, 1)
def okay(msg): return _nagios('OKAY', msg, 0)

def main(args):
  status = os.popen('riak ping').read()
  if status == "pong\n":
    okay("It's up!")
  else:
    critical("It's down!")
  
if __name__ == '__main__':
  sys.exit(main(sys.argv[0:]))

File: check_riak_up.erl
#!/usr/bin/env escript

main(_) ->
    Status = string:tokens(os:cmd("riak ping"), "\n"),
    case lists:any(fun(X) -> X == "pong" end, Status) of
        true -> nagios:okay("It's up!");
        _ -> nagios:critical("It's down!")
    end.
